WinUIB2CApp/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,248 @@
+using Microsoft.IdentityModel.Abstractions;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WinUIMSALAppB2C.MSAL
+{
+    /// <summary>
+    /// Contains methods that initialize and use the MSAL SDK
+    /// </summary>
+    /// <autogeneratedoc />
+    public class MSALClientHelper
+    {
+        public IntPtr windowHandle { get; set; }
+
+        /// <summary>
+        /// As for the Tenant, you can use a name as obtained from the azure portal, e.g. kko365.onmicrosoft.com"
+        /// </summary>
+        public readonly AzureADB2CConfig AzureADB2CConfig;
+
+        /// <summary>
+        /// Gets the authentication result (if available) from MSAL's various operations.
+        /// </summary>
+        /// <value>
+        /// The authentication result.
+        /// </value>
+        public AuthenticationResult AuthResult { get; private set; }
+
+        /// <summary>
+        /// Gets the MSAL public client application instance.
+        /// </summary>
+        /// <value>
+        /// The public client application.
+        /// </value>
+        public IPublicClientApplication PublicClientApplication { get; private set; }
+
+        private PublicClientApplicationBuilder PublicClientApplicationBuilder;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MSALClientHelper"/> class.
+        /// </summary>
+        public MSALClientHelper(AzureADB2CConfig azureADB2CConfig)
+        {
+            this.AzureADB2CConfig = azureADB2CConfig;
+
+            this.InitializePublicClientApplicationBuilder();
+        }
+
+        /// <summary>
+        /// Initializes the MSAL's PublicClientApplication builder.
+        /// </summary>
+        /// <autogeneratedoc />
+        private void InitializePublicClientApplicationBuilder()
+        {
+            this.PublicClientApplicationBuilder = PublicClientApplicationBuilder.Create(AzureADB2CConfig.ClientId)
+                .WithExperimentalFeatures() // this is for upcoming logger
+                .WithB2CAuthority($"{AzureADB2CConfig.Instance}/tfp/{AzureADB2CConfig.Domain}/{AzureADB2CConfig.SignUpSignInPolicyid}")
+                .WithRedirectUri($"msal{AzureADB2CConfig.ClientId}://auth")
+                .WithLogging(new IdentityLogger(EventLogLevel.Warning), enablePiiLogging: false)    // This is the currently recommended way to log MSAL message. For more info refer to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging. Set Identity Logging level to Warning which is a middle ground
+                .WithIosKeychainSecurityGroup("com.microsoft.adalcache");
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IAccount> InitializePublicClientAppAsync()
```

Comment:
`InitializePublicClientAppAsync` should not return accounts. It should return `IPublicClientApplication`